packages/embed-services/src/EmbedUrl.spec.ts

```diff
@@ -0,0 +1,158 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2023 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import { EmbedUrl, ContentType } from './EmbedUrl'
+
+describe('EmbedUrl', () => {
+  const dashboardUrl = 'https://example.com/dashboards/123'
+  const dashboardEmbedUrl = 'https://example.com/embed/dashboards/123'
+  const exploreUrl = 'https://example.com/explore/456'
+  const lookUrl = 'https://example.com/embed/looks/789'
+  const invalidUrl = 'https://example.com'
+
+  describe('constructor', () => {
+    it('should set searchParams and path properties if present', () => {
+      const url = `${dashboardUrl}?param1=value1`
+      const targetUrl = new EmbedUrl(url)
+      expect(targetUrl.searchParams).toEqual({ param1: 'value1' })
+      expect(targetUrl.path).toBe('/dashboards/123')
+    })
+
+    it('should support a target url without search params', () => {
+      const targetUrl = new EmbedUrl(lookUrl)
+      expect(targetUrl.searchParams).toEqual({})
+      expect(targetUrl.path).toBe('/embed/looks/789')
+    })
+
+    it('should correctly determine isDashboard, isExplore, and isLook properties', () => {
+      const dashboardTargetUrl = new EmbedUrl(dashboardUrl)
+      const exploreTargetUrl = new EmbedUrl(exploreUrl)
+      const lookTargetUrl = new EmbedUrl(lookUrl)
+
+      expect(dashboardTargetUrl.isDashboard).toBe(true)
+      expect(dashboardTargetUrl.isExplore).toBe(false)
+      expect(dashboardTargetUrl.isLook).toBe(false)
+
+      expect(exploreTargetUrl.isDashboard).toBe(false)
+      expect(exploreTargetUrl.isExplore).toBe(true)
+      expect(exploreTargetUrl.isLook).toBe(false)
+
+      expect(lookTargetUrl.isDashboard).toBe(false)
+      expect(lookTargetUrl.isExplore).toBe(false)
+      expect(lookTargetUrl.isLook).toBe(true)
+    })
+
+    it('should throw if target url is invalid', () => {
+      expect(() => new EmbedUrl(invalidUrl)).toThrowError(
+        'Unknown content type'
+      )
```

Comment:
Nice test pattern, but I'm wondering if we should throw an error for a utility class that tells us whether or not a given URL is an embed url

Reply:
Any ideas? I could set `contentType` to `invalid` and not error

Reply:
I like the idea of Invalid

Reply:
Alright. I'll do that and make `EmbedUrl.embedUrl()` throw if called

Reply:
Good idea